### DISCUSSION TITLE
Should 'reference frame' (and epoch) apply to more than just GNSS-type measurements?

### DISCUSSION THREAD

**Discussion by jhaasdyk-au** [GeoscienceAustralia/DynAdjust#198] [Ideas]
The purpose of this github discussion is to:
- record findings, feedback and questions to inform planned discussions with icsm-au/DynAdjust subject matter experts
   i.e. to inform additional user(self)-education, or potential enhancements or defects.
 
All findings and recommendations are my own and may not correctly represent current or planned DynAdjust behaviour.

---

**Should reference Frame (and epoch) apply to more than just GNSS-type measurements?
Or does it already?**

Currently the reference frame attribute is ONLY defined for GNSS-type measurements, per Section B.1.4 (with GNSS msr at Table B.8).
While GNSS are the most obvious observation types that rely on reference frame (and epoch), should reference frame (and epoch) also be applied for: 
- geodetic latitude (P), geodetic longitude (Q), astronomic latitude (I) astronomic longitude (J), orthometric height (H) and ellipsoid height (R), ellipsoid chord (C) and arc (E) distances?
- Others?   

I cannot find reasoning / discussion in the Users guide to support or explain the inclusion / exclusion of other MSR types.
I would benefit from that being recorded.

Note: Station coordinates of course also employ  reference frame and epoch (but in STN file header only, not per coordinate row). Out of scope for this discussion. 

---

**BACKGROUND:**
Various sections in the Users Guide address this concept:

Section 7.2. Observation equations
- “GNSS baselines (G) and baseline clusters (X), GNSS point clusters (Y), geodetic latitude (P) and longitude (Q) and ellipsoid height (R) are aligned with the adjustment ellipsoid, reference frame and epoch.
If a different ellipsoid, reference frame and/or epoch has been provided with such measurements, DynAdjust applies a transformation (as per §4.3) to align each measurement to the adjustment reference frame.”

- However, it is notable that the users guide describes the use of reftran on GNSS-type measurements only. 
I have not undertaken testing to date of its ACTUAL performance on other measurement types, but that could be instructive.
<br/>

Section 3.2.3 Network Constraints / Point measurement constraints.
- “The measurement types which act as positional constraints include GNSS point cluster (Y), geodetic latitude (P), geodetic longitude (Q), astronomic latitude (I) astronomic longitude (J), orthometric height (H) and ellipsoid height (R).

Various other sections refer to GNSS msr only
- Section 4.6 : “The primary purpose of reftran is to align all station coordinates and GNSS measurements to a common reference frame and epoch prior to least squares adjustment.”
- Section A.3: “reftran [definition:] Basic usage To transform imported station coordinates and GNSS measurement into a common reference frame using reftran”
- Section 3.3.1 Reference Frame: “DynAdjust provides a capability to transform station coordinates and GNSS measurements between several static and dynamic reference frames via reftran (see Chapter 4)”
“To this end, reftran expects all stations and measurements to be tagged with a recognised reference frame abbreviation [and epoch as required]”
" Any GNSS measurement which does not have a reference frame abbreviation assigned to it will adopt [the default reference frame upon import]"
- Section B.1.2 (Re information in Header line) “For measurement files, the default reference frame and epoch [provided in the header] will be overridden by the reference frame and epoch provided for each GNSS measurement”
(what currently happens with other measurements providing epoch?)

... while other sections refer more generically to measurements:
-	Section 1.3.1 (Software architecture and information flow): “reftran [definition:] This program performs reference frame transformations to align all imported stations and measurements to a common reference frame and epoch. 
-	Section 1.3.3 (Program execution sequence):  “reftran [definition:]  Transform station coordinates and measurements onto the specified reference frame and epoch.”
Which then goes on to describe:
      - “[for an example with] Multiple reference frames:  If the GNSS measurements in this survey are aligned to different refer ence frames, then the sequence import, **reftran** and adjust will be required. 
      -  “[for an example with] Terrestrial measurements: If terrestrial measurements (e.g. angles, distances and orthometric height differences) form part of this survey, then the sequence will change to import, **reftran**, geoid and adjust.

**Comment by rogerfraser** [comment]
Thanks @jhaasdyk-au for raising this question.  In my mind, "yes" would be the appropriate answer to your question given that all measurements are expressed in terms of a datum, whether recorded or not, be it astronomic, geodetic, local or height.  Similarly for epoch (c.f. your question, #195), noting also that epoch is a prerequisite for handling change via deformation model. 

But which measurement types?

As you've rightly noted, the mention of reference frame in the documentation is discussed almost entirely from the perspective of handling GNSS type measurements (G, Y, Y).  Height measurements (H, R, L) are perhaps the only exception and in this context, "datum" is the the more popular term even though it carries the same idea as reference frame.  Measurements in the local reference frame (A, D, S, V, Z) don't really enter this discussion, in my mind, since they are parameterised relative to the cartesian reference frame via the ellipsoid specifications of the default (or user-specified) reference frame and the geodetic location of the instrument station.

That leaves measurement types B, C, E, I, J, K, P, Q.  For each of these, the answer to your question will need to take into consideration (recognising datum might be the more practical term):
- Is a reference frame transformation applicable?
- Is re-reduction to the ellipsoid applicable?
- When might a deformation model applicable?
- Is the measurement observed?  Or is it computed/reduced?  If the latter, via measurement or via coordinates (whether geodetic or astronomic)?
- When is epoch relevant to transforming or reducing such measurements?  

There might be other questions to consider also
